isotp/examples/echo/main/isotp_echo_main.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Unlicense OR CC0-1.0
+ */
+
+#include <stdio.h>
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "esp_log.h"
+#include "esp_check.h"
+#include "esp_twai.h"
+#include "esp_twai_onchip.h"
+#include "esp_isotp.h"
+
+static const char *TAG = "isotp_echo";
+
+// Global variables for cleanup
+static esp_isotp_handle_t g_isotp_handle = NULL;
+static twai_node_handle_t g_twai_node = NULL;
+static QueueHandle_t g_rx_queue = NULL;
+
+static esp_err_t isotp_echo_init(void);
+static esp_err_t isotp_echo_deinit(void);
+
+void app_main(void)
+{
+    ESP_LOGI(TAG, "ISO-TP Echo Demo started");
+
+    esp_err_t ret = isotp_echo_init();
+    ESP_GOTO_ON_ERROR(ret, err, TAG, "Failed to initialize ISO-TP echo: %s", esp_err_to_name(ret));
+
+    while (1) {
+        vTaskDelay(pdMS_TO_TICKS(10000));
+    }
+
+    ESP_LOGI(TAG, "ISO-TP Echo Demo finished");
+
+err:
```

Comment:
we don't do complicated error handling in our example, we recommend to panic on any error. Suggest using "assert" and "ESP_ERROR_CHECK".